Sources/Crust/Shine.swift

```diff
@@ -0,0 +1,113 @@
+/// Parser.swift
+///
+/// Copyright 2017, The Silt Language Project.
+///
+/// This project is released under the MIT license, a copy of which is
+/// available in the repository.
+import Lithosphere
+
+private enum LayoutBlock {
+  case Implicit
+  case Explicit
+
+  var isImplicit : Bool {
+    switch self {
+    case .Implicit: return true
+    default: return false
+    }
+  }
+}
+
+/// Process a raw Silt token stream into a Stainless Silt token stream by
+/// inserting layout markers in the appropriate places.  This ensures that we
+/// have an explicitly-scoped input to the Parser before we even try to do a
+/// Scope Check.
+public func layout(_ ts : [TokenSyntax]) -> [TokenSyntax] {
+  var toks = ts
+  var stainlessToks = [TokenSyntax]()
+  var layoutBlockStack = [LayoutBlock]()
+  while !toks.isEmpty && toks[0].tokenKind != .eof {
```

Comment:
`while let tok = toks.removeFirst(), tok.tokenKind != .eof`?

Reply:
`removeFirst()` traps, it doesn't return an optional

Reply:
Ah, garbage. Cool, this is fine then.